test/e2e/knative_serving_test.go

```diff
@@ -301,3 +303,45 @@ func waitForRouteServingText(t *testing.T, caCtx *test.Context, routeDomain, exp
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeDomain, expectedText, err)
 	}
 }
+
+func testKnativeServingForGlobalProxy(t *testing.T, caCtx *test.Context) {
+	if err := test.UpdateGlobalProxy(caCtx, "http://1.2.4.5:8999"); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	t.Run("wait for it to be ready after update", func(t *testing.T) {
```

Comment:
I'm not sure all these subtests is how testing in golang is supposed to work. Can we remove all the subtests from this function?

Reply:
do you mean we need to have separate function for each t.Run ?

Reply:
not necessarily, why not just have all of the code flat?